includes/KSES.php

```diff
@@ -86,17 +97,25 @@ public static function get_registration_action_priority(): int {
 	 * @return array|mixed Filtered post data.
 	 */
 	public function filter_insert_post_data( $data, $postarr, $unsanitized_postarr ) {
+		if ( current_user_can( 'unfiltered_html' ) ) {
+			return $data;
+		}
+
 		if (
-			( Story_Post_Type::POST_TYPE_SLUG !== $data['post_type'] ) && !
+			( $this->story_post_type::POST_TYPE_SLUG !== $data['post_type'] ) && !
 			(
 				'revision' === $data['post_type'] &&
 				! empty( $data['post_parent'] ) &&
-				Story_Post_Type::POST_TYPE_SLUG === get_post_type( $data['post_parent'] )
+				get_post_type( $data['post_parent'] ) === $this->story_post_type::POST_TYPE_SLUG
 			)
 		) {
 			return $data;
 		}
 
+		if ( ! $this->get_post_type_cap( $this->story_post_type::POST_TYPE_SLUG, 'edit_posts' ) ) {
```

Comment:
We do this now and remove the trait on this class.
```suggestion
		if ( ! $this->story_post_type->get_post_type_cap( $this->story_post_type::POST_TYPE_SLUG, 'edit_posts' ) ) {
```

Reply:
We should probably get rid of this trait eventually when we can just access the instance..

Reply:
Also, I can't do this change because `get_post_type_cap` is `protected`

Reply:
Let's do this in a follow up PR.